R/req-promise.R

```diff
@@ -69,6 +73,12 @@ req_perform_promise <- function(req,
   check_installed(c("promises", "later"))
   check_string(path, allow_null = TRUE)
 
+  if (missing(pool)) {
```

Comment:
```suggestion
  if (is.null(pool)) {
```

(Since missingness isn't transitive across calls, and that makes it bit more clear what happens when pool isn't missing)

Reply:
But looking at the tests suggests that an explicit `NULL` is ok?

Reply:
Why it's `missing()` is that `NULL` is the only way to specify the default pool for `curl`. `curl:::multi_default()` which returns the default pool is an internal function.

From the discussion and @jcheng5's comment here: https://github.com/shikokuchuo/httr2/pull/1#issuecomment-2466824678, I think it'd be ok if that missingness turned into `NULL` and was accepted. It was meant more as a guard against mistakes rather than a failsafe. We do also say in the docs that a new pool should be created for private loops.

Reply:
Or if it's `NULL`, should we just automatically create a new pool instead of using the curl default pool?

Reply:
I thought about this, but
(i) I'm not sure you'd want `NULL` to mean different things in different contexts, even if documented, and
(ii) it would work if you only did one request in a private loop, otherwise it would inefficiently create a new pool for each request.

The pattern we want users to be using should be something like:
```r
later::with_temp_loop(
  pool <- curl::new_pool()
  r1 <- req_perform_promise(pool = pool)
  r2 <- req_perform_promise(pool = pool)
)
```

Reply:
Oh gotcha.

Reply:
Let's leave it as is then